core/src/Repository/DataSource/Sql/DataSource/RawSqlDataSource.php

```diff
@@ -131,6 +138,25 @@ public function put(Query $query, mixed $entity = null): mixed {
     return $this->get(new IdQuery($id));
   }
 
+  /**
+   * @throws QueryNotSupportedException
+   * @throws DataNotFoundException
+   */
+  public function queryPut(SqlQuery $query): mixed {
+    switch (true) {
+      case $query instanceof InsertSqlQuery:
+        $sql = $this->sqlBuilder->insert(null, $query);
+        $id = $this->pdo->insert($sql->sql(), $sql->params());
+        return $this->get(new IdQuery($id));
+      case $query instanceof PatchSqlQuery:
+        $sql = $this->sqlBuilder->patch($query);
+        $this->pdo->execute($sql->sql(), $sql->params());
```

Comment:
The update by value only works with id condition, new patch query inherits from WhereQuery so any condition is accepted